packages/react-scripts/config/modules.js

```diff
@@ -130,7 +134,39 @@ function getModules() {
     // Otherwise we'll check if there is jsconfig.json
     // for non TS projects.
   } else if (hasJsConfig) {
-    config = require(paths.appJsConfig);
+    if (hasTsSetup) {
+      const ts = require(resolve.sync('typescript', {
+        basedir: paths.appNodeModules,
+      }));
+
+      config = ts.readConfigFile(paths.appTsConfig, ts.sys.readFile).config;
+    } else {
+      try {
+        config = require(paths.appJsConfig);
+      } catch (err) {
+        console.error(
+          chalk.bold.red(
+            `It looks like your jsconfig.json file has comments but ${chalk.bold(
+              'typescript'
+            )} is not installed.`,
+            `If you want to support JSONC you have to install ${chalk.bold(
+              'typescript'
+            )} as a dependency.`
+          )
+        );
+        console.log(
+          chalk.bold(
+            'Please install',
+            chalk.cyan.bold('typescript'),
+            'by running',
+            chalk.cyan.bold(
+              isYarn ? 'yarn add typescript' : 'npm install typescript'
+            ) + '.'
```

Comment:
```suggestion
            ) + 'typescript'
```

Reply:
In this case `typescript` will be out of `chalk.cyan.bold`. I move ` typescript` as a separate string from both strings but still inside the `chalk`. Please let me know if it's not ok.

Reply:
I understand, and that makes sense. You can leave it, thanks.